cpp/src/io/parquet/page_decode.cuh

```diff
@@ -1159,13 +1159,15 @@ inline __device__ bool setupLocalPageInfo(page_state_s* const s,
           if (s->col.ts_clock_rate) {
             int32_t units = 0;
             // Duration types are not included because no scaling is done when reading
-            if (s->col.converted_type == TIMESTAMP_MILLIS) {
-              units = cudf::timestamp_ms::period::den;
-            } else if (s->col.converted_type == TIMESTAMP_MICROS) {
-              units = cudf::timestamp_us::period::den;
-            } else if (s->col.logical_type.has_value() and
-                       s->col.logical_type->is_timestamp_nanos()) {
-              units = cudf::timestamp_ns::period::den;
+            if (s->col.logical_type.has_value()) {
+              auto const& lt = s->col.logical_type.value();
```

Comment:
I've a silly question:  Is `LogicalType` expensive to copy?  (I ask since `lt` is `const&`.)

I couldn't make out why it would be.

Reply:
Well, it is a struct with a few optional members, but it wouldn't be ridiculously expensive to copy. I'm just trying to save typing `s->col.logical_type->is_timestamp_xxx()` a bunch of times, but could be convinced to do so.

Reply:
Not important at all. Just curious.